web/modules/custom/portland/portland.tokens.inc

```diff
@@ -127,6 +132,29 @@ function portland_tokens($type, $tokens, array $data = array(), array $options =
             }
           }
           break;
+        case 'group-shortname':
+          // get group shortname
+          $route_match = \Drupal::routeMatch();
+          // Entity will be found in the route parameters.
+          if (($route = $route_match->getRouteObject()) && ($parameters = $route->getOption('parameters'))) {
+            // Determine if the current route represents an entity.
+            foreach ($parameters as $name => $options) {
+              if ($name == 'group') {
+                if (isset($options['type']) && strpos($options['type'], 'entity:') === 0) {
+                  $entity = $route_match->getParameter($name);
+                  if ($entity) {
+                    $field = $entity->get('field_shortname_or_acronym');
```

Comment:
Hmm. There is a catch here. Shortname or Acronym is not required. Maybe we should use field_group_path instead. If we do, can we make it a safe token that adds the extra directory for groups that are within the path of their parent?